### PR TITLE
Drop SQLite support

### PR DESCRIPTION
**Pull request by Dzejkop**
From now on, we only support Posgtres.

Each test will start up a postgres docker container.

Note that this forces us to have docker installed.

### CODE REVIEW

File: crates/docker-utils/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    }
+}
+
+/// Starts a postgres docker container that will accept all connections with a
+/// random port assigned by the os or docker. The container will be stopped and
+/// removed when the guard is dropped.
+///
+/// Note that we're using sync code here so we'll block the executor - but this
+/// is fine, because the spawned container will still run in the background.
+pub async fn setup() -> anyhow::Result<DockerContainerGuard> {
```
Comment (iamrecursion) [inline crates/docker-utils/src/lib.rs:38]: Given the crate is called `docker-utils`, this name isn't descriptive enough. Either the crate should be something like `pg-docker-utils`, or the function name should be more descriptive.

**Review by iamrecursion — APPROVED** [review]
Good stuff, some nits inline, and please fix the clippy lints.

File: crates/docker-utils/src/lib.rs (hunk 1 above)
Comment (Dzejkop) [reply crates/docker-utils/src/lib.rs]: Renamed to `postgres-docker-utils`

File: tests/integration_tests.rs [hunk 2]
```diff
@@ -51,15 +50,16 @@ async fn validate_proofs() {
     init_tracing_subscriber();
     info!("Starting integration test");
 
-    let db_dir = tempdir().unwrap();
-    let db = db_dir.path().join("test.db");
+    let db_container = docker_utils::setup().await.unwrap();
+    let port = db_container.port();
+    let db_url = format!("postgres://postgres:postgres@localhost:{port}/database");
```
[the remark this answers is not in the document]
Comment (Dzejkop) [reply tests/integration_tests.rs]: Added a section in README about dependencies

File: src/database/mod.rs
[the remark this answers is not in the document]
Comment (Dzejkop) [reply src/database/mod.rs:1]: I revisited the queries and changed the one you left a comment with. I looked that query with the 2 nested queries (the one in `get_root_state`) but it looks fine to me.

**Review by iamrecursion — APPROVED** [review]
LGTM!

**Pull request merged**